openmdao/recorders/case_recorder.py

```diff
@@ -164,25 +187,22 @@ def record_iteration(self, recording_requester, data, metadata, **kwargs):
         **kwargs : keyword args
             Some implementations of record_iteration need additional args.
         """
-        if not self._parallel:
-            if MPI and MPI.COMM_WORLD.rank > 0:
-                raise RuntimeError("Non-parallel recorders should not be recording on ranks > 0")
-
-        self._counter += 1
-
-        self._iteration_coordinate = \
-            recording_requester._recording_iter.get_formatted_iteration_coordinate()
-
-        if isinstance(recording_requester, Driver):
-            self.record_iteration_driver(recording_requester, data, metadata)
-        elif isinstance(recording_requester, System):
-            self.record_iteration_system(recording_requester, data, metadata)
-        elif isinstance(recording_requester, Solver):
-            self.record_iteration_solver(recording_requester, data, metadata)
-        elif isinstance(recording_requester, Problem):
-            self.record_iteration_problem(recording_requester, data, metadata)
-        else:
-            raise ValueError("Recorders must be attached to Drivers, Systems, or Solvers.")
+        if self.record_on_proc:
```

Comment:
Just to be clear, _Undefined triggers this condition, and that's expected? (As in the case where we have multiple procs but aren't running parallel, and the distributed drivers set all other procs to False, but don't set this one.)

Reply:
right.. this is a bug.. need a test.

Reply:
reworked the logic and got rid of _UNDEFINED